src/wasm.h

```diff
@@ -1003,6 +1007,52 @@ class Pop : public SpecificExpression<Expression::PopId> {
   Pop(MixedArena& allocator) {}
 };
 
+class Try : public SpecificExpression<Expression::TryId> {
+public:
+  Try(MixedArena& allocator) {}
+
+  Expression* body;
+  Expression* catchBody;
+
+  void finalize();
+  void finalize(Type type_);
+};
+
+class Throw : public SpecificExpression<Expression::ThrowId> {
+public:
+  Throw(MixedArena& allocator) : operands(allocator) {}
+
+  Name event;
+  ExpressionList operands;
+
+  void finalize();
+};
+
+class Rethrow : public SpecificExpression<Expression::RethrowId> {
+public:
+  Rethrow(MixedArena& allocator) {}
+
+  Expression* exnref;
+
+  void finalize();
+};
+
+class BrOnExn : public SpecificExpression<Expression::BrOnExnId> {
+public:
+  BrOnExn() { type = unreachable; }
+  BrOnExn(MixedArena& allocator) : BrOnExn() {}
+
+  Name name;
+  Name event;
+  Expression* exnref;
+  // This is duplicate info of param types stored in Event, but this is required
+  // when BrOnExn expression is refinalized independently without modules.
+  std::vector<Type> eventParams;
+
+  void finalize();
+  Type getSingleEventType();
```

Comment:
maybe `getSentType()` or `getSingleSentType()`? As in, "value sent to the block"?

Reply:
Fixed to `getSingleSentType`, because actually we might be supposed to send multiple types if the event type contains multiple types. 'Single' here means something like this is a temporary solution before multi value support.